### PR TITLE
Temporary fix for #854

### PR DESCRIPTION
**Pull request by summivox**
`[.. for [to 10]]` is wrongly rewritten by lexer into `[.. for [from 0 to 10]]` (not quite, see below). Patching the lexer could be non-trivial as said rewriting is currently ad-hoc (`forange` in `lexer.ls`, see #854 for discussion). This temporary solution relaxes the grammar and allow the malformed lexer output.

Caveat: `[from x to y]` is *not* supported directly from source despite the change in grammar due to the same lexer hack. The `from` is currently considered a literal rather than a `FROM` token.

A possibly cleaner solution is to get rid of `forange` and support `from` elision in a separate lexer rewrite step or in parser instead.

### CODE REVIEW

File: src/grammar.ls [hunk 1]
```diff
@@ -422,7 +426,7 @@ bnf =
         o 'FOR ID FROM Expression TO Expression BY Expression'
         , -> new For kind: $1, index: $2, from: $4, op: $5, to: $6, step: $8
         o 'FOR FROM Expression TO Expression BY Expression'
-        , -> new For kind: $1,            from: $3, op: $4, to: $5, step: $6, ref: true
+        , -> new For kind: $1,            from: $3, op: $4, to: $5, step: $7, ref: true
```
Comment (vendethiel) [inline src/grammar.ls]: Not sure why this commit isn't considered merged by github...

**Comment by vendethiel** [comment]
This is far too many FOR variations... It should be refactored with some OptBy/... - also, the change should probably allow us go get rid of forange completly. That **should not** be in the lexer.

**Comment by summivox** [comment]
Agreed.

There is one potential show-stopper though: In current specification `from`, `to`/`til`, `by` are so-called "contextual keywords". This makes it difficult to detect literal ranges without having the lexer do way more parsing than it should. For example:
```livescript
x = [(to til) to (by from)]
```
which is sick :joy: but sufficiently points out that "contextual keywords" are neither well-defined nor well-implemented. 

My opinion is to make at least `to`, `til`, and `by` reserved (since we are clearly talking about LS2.0 here we can stop worrying about sick cases). There has been quite some discussion on keyword usage (e.g. #818) and the conclusion seems to favor "escape yourself if you know what you're doing, otherwise don't even bother".

**Comment by vendethiel** [comment]
The example you linked isn't exact an issue, because the per-nesting flags should deal with that.

But yes. let's make them reserved.

**Comment by vendethiel** [comment]
@summivox would you mind making them reserved (should be easy, just need to add them to the keywords array), or do you want me to do it?

**Comment by summivox** [comment]
I'm busy with my masters thesis these days. Could you please do it?

**Pull request merged**